lib/cfg-lexer.c

```diff
@@ -935,18 +935,18 @@ cfg_lexer_lex(CfgLexer *self, YYSTYPE *yylval, YYLTYPE *yylloc)
     }
   else if (configuration->user_version == 0 && configuration->parsed_version != 0)
     {
-      cfg_set_version(configuration, configuration->parsed_version);
+      if (!cfg_set_version(configuration, configuration->parsed_version))
+        return LL_ERROR;
     }
   else if (cfg_lexer_get_context_type(self) != LL_CONTEXT_PRAGMA && !self->non_pragma_seen)
     {
       /* first non-pragma token */
 
       if (configuration->user_version == 0 && configuration->parsed_version == 0)
         {
-          /* no version selected yet, and we have a non-pragma token, this
-           * means that the configuration is meant for syslog-ng 2.1 */
-          msg_warning("WARNING: Configuration file has no version number, assuming syslog-ng 2.1 format. Please add @version: maj.min to the beginning of the file to indicate this explicitly");
-          cfg_set_version(configuration, 0x0201);
+          msg_error("ERROR: configuration files without a version number has become unsupported in " VERSION_3_13
+                    ", please specify a version number using @version and update your configuration accordingly");
+          return LL_ERROR;
```

Comment:
Wouldn't be better if we fallback to the latest version (with a warning message) instead of returning with this error?

Reply:
We talked about it IRL and we keep this as is.

Reply:
Do you happen to remember the argument behind that?
Asking, because using the latest version might be a solution to #3068